private_sharing/models.py

```diff
@@ -655,7 +656,7 @@ class ProjectDataFile(DataFile):
         related_name="parent_project_data_file",
         on_delete=models.CASCADE,
     )
-
+    datatypes = models.ManyToManyField(DataTypes, related_name="datafile_datatypes")
```

Comment:
how about `registered_datatype`? then you can skip the related_name, I think the default would be `project.registered_datatype_set`

Note... there isn't any `requested_datatype` yet but something like that will also exist (although notably sometimes the datatypes will have project-specificity, implementation TBD...)